components/ipfs/translate_ipfs_uri.cc

```diff
@@ -42,13 +46,24 @@ bool TranslateIPFSURI(const GURL& url, GURL* new_url, const GURL& gateway_url) {
       // new_url would be:
       // https://dweb.link/ipfs/[cid]//wiki/Vincent_van_Gogh.html
       if (new_url) {
-        std::string gateway_empty_path = gateway_url.spec();
-        if (gateway_empty_path.size() > 0 &&
-            gateway_empty_path[gateway_empty_path.size() - 1] == '/') {
-          gateway_empty_path.pop_back();
+        if (use_subdomain) {
+          GURL uri(gateway_url);
+          GURL::Replacements replacements;
+          std::string host = base::StringPrintf("%s.%s.%s", cid.c_str(),
+                                                ipfs_scheme ? "ipfs" : "ipns",
+                                                gateway_url.host().c_str());
+          replacements.SetHostStr(host);
+          replacements.SetPathStr(path);
+          *new_url = gateway_url.ReplaceComponents(replacements);
+        } else {
+          std::string gateway_empty_path = gateway_url.spec();
+          if (gateway_empty_path.size() > 0 &&
+              gateway_empty_path[gateway_empty_path.size() - 1] == '/') {
```

Comment:
this is `gateway_empty_path.back()`

Reply:
that code is removed in the rebase on master I'll be pushing soon, so no worries.